src/arrayMethodReduce.js

```diff
@@ -4,8 +4,16 @@
  * Implement method Reduce
  */
 function applyCustomReduce() {
-  [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+  [].__proto__.reduce2 = function(
+    callback,
+    initialValue = typeof this[0] === 'number' ? 0 : '') {
```

Comment:
```suggestion
    initialValue = typeof this[0] === 'number' ? 0 : '') {
```
ні там не так працює, якщо його не передали там буде перший елемент масиву виступаки акумулятором